### PR TITLE
sort nurses by task status of active participants

### PR DESCRIPTION
**Pull request by ericcf**
[#116371417]

### CODE REVIEW

File: app/controllers/nurse_supervisor_dashboards_controller.rb [hunk 1]
```diff
@@ -2,13 +2,15 @@
 # Manage Nurse Supervisor display.
 class NurseSupervisorDashboardsController < ApplicationController
   def show
-    participants = Participant.where(nurse: current_user.nurses)
+    participants = Participant.where(nurse: current_user.nurses).includes(:nurse)
```
Comment (michaelwehrley) [inline app/controllers/nurse_supervisor_dashboards_controller.rb]: How come you decided to eagerly load these?  Was this an optimization?

File: app/controllers/nurse_supervisor_dashboards_controller.rb [hunk 2]
```diff
@@ -2,13 +2,15 @@
...
     @pending_participants = Participant.pending
                                        .where(locale: current_user.locale)
+                                       .includes(:nurse)
     @active_participants = participants.active
     @completed_participants = participants.completed
     @dropped_out_participants = participants.dropped_out
-    @nurses = current_user.nurses
+    @dashboard = NurseSupervisorDashboardPresenter.new(current_user,
```
Comment (michaelwehrley) [inline app/controllers/nurse_supervisor_dashboards_controller.rb]: Next time, you could go even further and have a presenter that creates just one instance variable.

File: app/controllers/nurse_supervisor_dashboards_controller.rb (hunk 1 above)
Comment (ericcf) [inline app/controllers/nurse_supervisor_dashboards_controller.rb]: Yes, the nurse data is needed in the view.

File: spec/presenters/nurse_supervisor_dashboard_presenter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+require "rails_helper"
+
+RSpec.describe NurseSupervisorDashboardPresenter do
```
Comment (michaelwehrley) [inline spec/presenters/nurse_supervisor_dashboard_presenter_spec.rb]: Nice spec.

File: app/controllers/nurse_supervisor_dashboards_controller.rb (hunk 2 above)
Comment (ericcf) [inline app/controllers/nurse_supervisor_dashboards_controller.rb]: Agreed. Next time it needs a refactor I'll probably do that.

**Comment by michaelwehrley** [comment]
Sorry I didn't get around to this, but great updates and use of presenter and specs. :rocket:

**Pull request merged**